File: mobile/lib/model/product_model.dart
class ProductModel {
  String mongoId;
  int productId;
  String name;
  String description;
  double price;
  String img;
  double rating;
  int stars;
  String createdAt;
  String updatedAt;

  ProductModel({
    required this.mongoId,
    required this.productId,
    required this.name,
    required this.description,
    required this.price,
    required this.stars,
    required this.img,
    required this.rating,
    required this.createdAt,
    required this.updatedAt,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      '_id': mongoId,
      'productId': productId,
      'name': name,
      'description': description,
      'price': price,
      'img': img,
      'rating': rating,
      'stars': stars,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
    };
  }

  factory ProductModel.fromMap(Map<String, dynamic> map) {
    return ProductModel(
      mongoId: map['_id'] as String,
      productId: map['productId'] as int,
      name: map['name'] as String,
      description: map['description'] as String,
      price: map['price'] as double,
      img: map['img'] as String,
      rating: map['rating'] as double,
      stars: map['stars'] as int,
      createdAt: map['createdAt'] as String,
      updatedAt: map['updatedAt'] as String,
    );
  }
}

File: server/model/product_model.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const productSchema = new Schema({
  productId: { type: Number, required: true },
  name: { type: String, required: true },
  description: { type: String },
  price: { type: Number, required: true },
  img: { type: String },
  rating: { type: Number },
  stars: { type: Number },
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now },
});

const ProductModel = mongoose.model('Product', productSchema);

module.exports = ProductModel;